### ISSUE TITLE
Help to add a learning rate scheduler

### ISSUE THREAD

**Issue by mathmanu** [ClementPinard/SfmLearner-Pytorch#5]
Hi, 

I tried to add a learning rate scheduler - file attached.  
[SfmLearner-Pytorch.zip](https://github.com/ClementPinard/SfmLearner-Pytorch/files/1477370/SfmLearner-Pytorch.zip)

It can be used in this way:
python3 train.py ./data/kitti/kitti_rawdata_formatted/ -b 48 -m 1.0 -s 0.1 --sequence-length 3 --log-output --checkpoint_dir ./training/checkpoints/ --optimizer Adam --scheduler step --lr 2e-4

However, the loss doesn't come down if I use the scheduler. Could you please take a look and let me know if I am doing something wrong.

**Comment by ClementPinard** [comment]
```python
if scheduler:
    scheduler.step()
else:
    optimizer.step()
```
scheduler.step() only changes learning rate, but does not perform iptimizer step. You also have to call optimizer.step(), regardless of the fact that you use scheduler or not.

```python
if scheduler:
    scheduler.step()
optimizer.step()
```

See source code here, to see why : http://pytorch.org/docs/master/_modules/torch/optim/lr_scheduler.html#StepLR

**Comment by mathmanu** [comment]
It's working now. Thank you.
